From 1cc5b4a9fb984e83681ae5c797fa6b22bc20f809
From: Ludovic Rousseau
Date: Fri, 16 Sep 2011 18:07:56 +0200
Subject: [PATCH] Correctly handle LIBUSB_TRANSFER_OVERFLOW in libusb_control_transfer()

sync.c: In function `libusb_control_transfer':
sync.c:122: warning: enumeration value `LIBUSB_TRANSFER_OVERFLOW' not
handled in switch

Fixes #120.
---
 libusb/sync.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/libusb/sync.c b/libusb/sync.c
index d50413b..8eed47b 100644
--- a/libusb/sync.c
+++ b/libusb/sync.c
@@ -132,6 +132,9 @@ int API_EXPORTED libusb_control_transfer(libusb_device_handle *dev_handle,
 	case LIBUSB_TRANSFER_NO_DEVICE:
 		r = LIBUSB_ERROR_NO_DEVICE;
 		break;
+	case LIBUSB_TRANSFER_OVERFLOW:
+		r = LIBUSB_ERROR_OVERFLOW;
+		break;
 	default:
 		usbi_warn(HANDLE_CTX(dev_handle),
 			"unrecognised status code %d", transfer->status);
-- 
1.7.6.1

